awx/api/serializers.py

```diff
@@ -4921,6 +4925,18 @@ class Meta:
         extra_kwargs = {
             'node_type': {'initial': Instance.Types.EXECUTION, 'default': Instance.Types.EXECUTION},
             'node_state': {'initial': Instance.States.INSTALLED, 'default': Instance.States.INSTALLED},
+            'hostname': {
+                'validators': [
+                    MaxLengthValidator(limit_value=255),
+                    RegexValidator(
+                        regex='^localhost$|^127.0.0.1$|^0000:0000:0000:0000:0000:0000:0000:0001$',
```

Comment:
https://stackoverflow.com/questions/8426171/what-regex-will-match-all-loopback-addresses has some ideas about how to capture the optional removal of various permutations of the zeros from the IPv6 localhost.